Updates for Terraform policy docs (#2864)
hashicorp/web-unified-docs: content/terraform-policy/v0.1.x (beta)/docs/policy/reference/test/provider.mdx

## Configuration model

The `provider` block supports the following configuration:

- `provider` &nbsp "[`<provider_name>`](#provider-name)" &nbsp "[`<test_label>`](#test-label)" &nbsp block
  - [`attrs`](#attrs) &nbsp object | optional
    - `<attribute_name>` &nbsp any | repeatable
  - [`expect_failure`](#expect-failure) &nbsp boolean | optional
  - [`inputs`](#inputs) &nbsp block | optional
    - `<input_name>` &nbsp any | repeatable
  - [`meta`](#meta) &nbsp object | optional
    - `alias` &nbsp string
    - `name` &nbsp string
    - `namespace` &nbsp string
    - `source` &nbsp string
    - `type` &nbsp string

>     - `version` &nbsp string

Reviewer: ```suggestion
  - [`attrs`](#attrs) &nbsp object | optional
    - `<attribute_name>` &nbsp any | repeatable
  - [`expect_failure`](#expect_failure) &nbsp boolean | optional
  - [`inputs`](#inputs) &nbsp block | optional
    - [`<input_name>`](#inputs) &nbsp any | repeatable
  - [`meta`](#meta) &nbsp object | optional
    - [`alias`](#meta) &nbsp string
    - [`name`](#meta) &nbsp string
    - [`namespace`](#meta) &nbsp string
    - [`source`](#meta) &nbsp string
    - [`type`](#meta) &nbsp string
    - [`version`](#meta) &nbsp string
```

## Complete configuration example

> The following example demonstrates a provider block with all configuration options specified:

Reviewer: ```suggestion
The following example demonstrates a `provider` block with all configuration options specified:
```

<CodeBlockConfig filename="tests/example.policytest.hcl">

```hcl
provider "aws" "all_options" {
  expect_failure = true
  
  meta = {
    type      = "aws"
    name      = "aws"
    source    = "hashicorp/aws"
    version   = "4.0.0"
    alias     = "secondary"
    namespace = "hashicorp"
  }
  
  attrs = {
    region = "us-east-1"
  }

  inputs {
    min_version = "5.0.0"
  }
}
```

</CodeBlockConfig>

## Specification

A `provider` block supports the following configuration.

### Provider name

> The provider name (For example: `aws`, `google`, `azurerm`).

Reviewer: ```suggestion
The provider name, such as `aws`, `google`, and `azurerm`.
```

| `source` | String | The provider source (For example: `"hashicorp/aws"`). |
| `version` | String | The provider version (For example: `"5.0.0"`). |
| `alias` | String | The provider alias, if testing aliased providers. |
| `name` | String | The provider's local name. |
| `namespace` | String | The provider registry namespace. |
| `type` | String | The provider type. |

## Examples

The following examples demonstrate common mock provider configuration patterns for specific use cases.

### Test provider source and version

> In the following example, mock `provider` blocks test a policy that validates provider source and version - one that will pass and one that will fail.

Reviewer: ```suggestion
In the following example, mock `provider` blocks test a policy that validates provider source and version. The first block mocks the passing scenario and the second mocks the failing scenario.
```

### Test provider configuration

> In the following example, mock `provider` blocks test a policy that enforces a specific AWS region - one with the correct region and one with an incorrect region.

Reviewer: ```suggestion
In the following example, mock `provider` blocks test a policy that enforces a specific AWS region. The first block mocks the correct region and the second block mocks an incorrect region.
```

    region = "us-west-2"
  }
  
  inputs {
    allowed_region = "us-west-2"
  }
}
```

### Test multiple provider configurations

> In the following example, mock `provider` blocks test policies across multiple provider types (AWS and Google) with different configurations.

Reviewer: ```suggestion
In the following example, mock `provider` blocks test policies across AWS and Google cloud providers with different configurations.
```
